### ISSUE TITLE
Travis Build Failing

### ISSUE THREAD

**Issue by matwright** [Laminas-Commons/LmcUser#28]
There are a number of notices and several fatal errors in the Test suite preventing the automated builds from suceeding.

`PHP Fatal error:  Declaration of LmcUserTest\Authentication\Adapter\TestAsset\AbstractAdapterExtension::authenticate(Laminas\EventManager\EventInterface $e) must be compatible with LmcUser\Authentication\Adapter\ChainableAdapter::authenticate(LmcUser\Authentication\Adapter\AdapterChainEvent $e) in /home/travis/build/Laminas-Commons/LmcUser/tests/LmcUserTest/Authentication/Adapter/TestAsset/AbstractAdapterExtension.php on line 8`

**Comment by visto9259** [comment]
We should probably remove testing with older versions of PHP and HHVM which are no longer supported.
And also bump up phpunit to a more recent supported version.  That's not a simple task since there are BC when migrating to more recent versions of phpunit.

**Comment by matwright** [comment]
I have completed this work. I've removed <7.3 as those versions are not supported in at least one of the Laminas components used in LmcUser or it's dependencies.

I've updated phpunit to the latest version (a jump from v3 to v9!) and corrected/updated many tests that were using since-removed phpunity methods.

My PR is passing Travis now, too.

https://github.com/Laminas-Commons/LmcUser/pull/30

If you'd like to review @visto9259 we can merge into the 3.1.x branch.

At that point I can look at merging the other PR into the branch and a new release asap.
